frontend/src/components/SpecialistsList/SpecialistsList.tsx

```diff
@@ -25,35 +31,40 @@ export const SpecialistsList: FunctionComponent<Props> = props => {
             .then(response => {
                 setLocations(response.data.locations);
             })
-            .catch(error => {
-                toast.error(error.message);
+            .catch(() => {
+                toast.error('Error: Could not get locations');
             });
     }, []);
 
     useEffect(() => {
         if (props.serviceId === undefined) return;
-        servicesService.getOfferedServices(props.serviceId, locationId, minimumRating).then(response => {
+
+        const locId = typeof locationId === 'string' ? undefined : locationId;
+
+        servicesService.getOfferedServices(props.serviceId, locId, minimumRating).then(response => {
             setOfferedServices(response.data.offeredServices);
         });
     }, [locationId, minimumRating, props.serviceId]);
 
     const locationChangeHandler = (data: DropdownProps) => {
         const value = data.value as number;
-        setLocationId(value);
+        setLocationId(+value);
```

Comment:
Why do you put '+' sign here ? It is already of a number type since you cast to number.